socketio/src/client/client.rs

```diff
@@ -322,6 +338,84 @@ mod test {
         Ok(())
     }
 
+    #[test]
+    fn socket_io_reconnect_url_auth_integration() -> Result<()> {
+        fn get_url() -> Url {
+            let timestamp = SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_millis();
+            let mut url = crate::test::socket_io_restart_url_auth_server();
+            url.set_query(Some(&format!("timestamp={timestamp}")));
+            url
+        }
+
+        let connect_num = Arc::new(AtomicUsize::new(0));
+        let close_num = Arc::new(AtomicUsize::new(0));
+        let message_num = Arc::new(AtomicUsize::new(0));
+
+        let connect_num_clone = Arc::clone(&connect_num);
+        let close_num_clone = Arc::clone(&close_num);
+        let message_num_clone = Arc::clone(&message_num);
+
+        let socket = ClientBuilder::new(get_url())
+            .reconnect(true)
+            .max_reconnect_attempts(100)
+            .reconnect_delay(100, 100)
+            .on(Event::Connect, move |_, socket| {
+                connect_num_clone.fetch_add(1, Ordering::SeqCst);
+                let result = socket.emit_with_ack(
+                    "message",
+                    json!(""),
+                    Duration::from_millis(100),
+                    |_, _| {},
+                );
+                assert!(result.is_ok(), "should emit message success");
+            })
+            .on(Event::Close, move |_, _| {
+                close_num_clone.fetch_add(1, Ordering::SeqCst);
+            })
+            .on("message", move |_, _| {
+                // test the iterator implementation and make sure there is a constant
+                // stream of packets, even when reconnecting
+                message_num_clone.fetch_add(1, Ordering::SeqCst);
+            })
+            .connect();
+
+        assert!(socket.is_ok(), "should connect success");
+        let socket = socket.unwrap();
+
+        // waiting for server to emit message
+        std::thread::sleep(std::time::Duration::from_millis(500));
```

Comment:
Please rather use a channel to signal you're done, this can lead to timing issues in CI.

Reply:
This is the same as the existing test. We have to wait a bit for the server to emit a message.

The (nodejs) server can't send a signal to the Rust code using a channel.

Reply:
Ok, then should be fine for now.